catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js

```diff
@@ -14,37 +14,227 @@
+        ich.addTemplate('newpreferencesModalTemplate', newpreferencesModalTemplate);
+        ich.addTemplate('colorPrefsTabTemplate', colorPrefsTabTemplate);
+        ich.addTemplate('layerPickerViewTemplate', layerPickerViewTemplate);
+        ich.addTemplate('layerPickerTemplate', layerPickerTemplate);
 
-        return Modal.extend({
-            template: 'preferencesModalTemplate',
+        var PrefsModalView = Modal.extend({
+            template: 'newpreferencesModalTemplate',
+            className: 'well well-small',
+            regions: {
+                colorTabRegion: "#colorTab",
+                layerTabRegion: "#layerTab"
+            },
+            events: {
+                'shown.bs.tab .tabs-below>.nav-tabs>li>a': 'tabShown',
+                'hidden.bs.modal': 'savePrefsChanges'
+            },
+            initialize: function () {
+                // work-around for newuser; to be removed when user "new" user model is working
+                this.model = new Newuser.Model();
+            },
+            tabShown: function (e) {
+                wreqr.vent.trigger('prefsModal:tabshown', e.target.hash);
+            },
+            onRender: function () {
+                this.colorTabRegion.show(new PrefsModalView.ColorTabView({model: this.model}));
+                this.layerTabRegion.show(new PrefsModalView.LayerTabView({model: this.model}));
+                wreqr.vent.trigger('prefsModal:tabshown', this.$('.nav-tabs > .active a').attr('href'));
+            },
+            savePrefsChanges: function () {
+                //this.model.get('user').get('user>colorPrefs').savePreferences();
+                this.model.get('colorPrefs').savePreferences();
+            }
+        });
+
+        PrefsModalView.LayerPicker = Marionette.ItemView.extend({
+            template: 'layerPickerTemplate',
+            tagName: 'tr',
+            className: 'layerPicker-row',
             events: {
-                'hidden.bs.modal': 'saveChanges'
+                'click input[type="range"]': 'changeAlpha',
+                'click input[type="checkbox"]': 'changeShow',
+                'click button[data-value="up"]': 'changeUp',
+                'click button[data-value="down"]': 'changeDown'
+            },
+            initialize: function (options) {
+                this.viewOptions = options;
+                Handlebars.registerHelper('canRaise', function () {
+                    var index = options.collection.indexOf(options.model);
+                    return index < options.collection.length - 1 ? '' : 'pickerNoRaise';
+                });
+                Handlebars.registerHelper('canLower', function () {
+                    var index = options.collection.indexOf(options.model);
+                    return index > 0 ? '' : 'pickerNoLower';
+                });
+            },
+            changeAlpha: function () {
+                var newAlphaValue = this.$el.find('input[type="range"]').val();
```

Comment:
Another short term tactic.  Yes, I will give the model binder a try.  The trick is to bind the cesium widget layers to the model collection, not the layer picker list.